.travis.yml

```diff
@@ -1,27 +1,35 @@
-sudo: false
+os: linux
 language: python
 cache: pip
-matrix:
+env: PYTHON=python
+jobs:
     include:
         # Linux
         - python: 2.7
-        - python: 3.7
-          dist: xenial
-          sudo: true
+        - python: 3.8
         # macOS
-        - language: generic
-          os: osx
-          env: PYVER=py27
-        - language: generic
-          os: osx
-          env: PYVER=py36
+        - os: osx
+          osx_image: xcode11.3
+          language: generic
+          env: PYTHON=python2.7
+        - os: osx
+          osx_image: xcode11.3
+          language: generic
+          env: PYTHON=python3.7
 install:
-    - ./.ci/travis/install.sh
+    - $PYTHON -m pip install --upgrade pip setuptools
+    - $PYTHON -m pip install --upgrade coveralls flake8 mock psutil pyopenssl pysendfile pytest six
```

Comment:
Does it make more sense to store these dependencies in e.g. a `test-requirements.txt` file?

Reply:
There are two philosophies...
1) Keep all that we need in a single file.  _or_
2) Spread state and commands across multiple files.

This PR was focused on 1) by moving state and commands into a single file and deleting three other files.

Do contributors run these same test locally?

Reply:
It was just a query, I'll leave the decision up to @giampaolo